src/managers/EventManager.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import type { EventPublisher } from 'spark-protocol';
+import type { DeviceAttributeRepository, Event, EventData } from '../types';
+
+class EventManager {
+  _eventPublisher: EventPublisher;
+  _deviceAttributeRepository: DeviceAttributeRepository;
+
+  constructor(
+    deviceAttributeRepository: DeviceAttributeRepository,
+    eventPublisher: EventPublisher,
+  ) {
+    this._deviceAttributeRepository = deviceAttributeRepository;
+    this._eventPublisher = eventPublisher;
+  }
+
+  _filterEvents = (
+    eventHandler: (event: Event) => void,
+    userID?: string,
+    deviceID?: string,
+  ): (event: Event) => Promise<void> =>
+    async (event: Event): Promise<void> => {
```

Comment:
hm.. just ` return;` works now, Flow was screaming at me about it before..now its ok.
"Is there any time when you would want to reject the promise instead?" - nope. I use async here just to be able to call
await on `this._deviceAttributeRepository.doesUserHaveAccess`,..if there are some errors, or event doesn't belong to the user, I just return, without calling eventHandler()..i don't need to throw anything etc here.

When you said weird pattern, are you about     `this._filterEvents(eventHandler, userID, deviceID),` - this in general?
but how can I filter incoming events another way?